### ISSUE TITLE
Scala 2.12 requires JRE 8

### ISSUE THREAD

**Issue by xpe** [NixOS/nixpkgs#22439]
### Underlying Cause

Scala 2.12 requires a version 8 JRE or JDK. (Java 7 will not work.) The current scala package does not reflect this dependency on Java 8.

### Symptom

Running scala fails:

```sh
% scala
Exception in thread "main" java.lang.UnsupportedClassVersionError: scala/tools/nsc/MainGenericRunner : Unsupported major.minor version 52.0
    at java.lang.ClassLoader.findBootstrapClass(Native Method)
    at java.lang.ClassLoader.findBootstrapClassOrNull(ClassLoader.java:1070)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:414)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:412)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
    at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)
```

The source of the problem is that JAVA_HOME (set by the Nix wrapper script) points to a version 7 JRE. See the source code is https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/scala/default.nix#L24

### Steps to reproduce

```sh
nix-env -i scala
scala
```

## My Workaround

Update `~/.nixpkgs/config.nix` to include:

```
{
  packageOverrides = pkgs: {
    scala = pkgs.scala.override { 
      jre = pkgs.jre8;
    };
  };
}
```

After this, `scala` works.

## Recommended Solution

Update the scala Nix package so the above user-level customization is not required.

## Technical details

* System: macOS 10.12.3
* Nix version: "nix-env (Nix) 1.11.6"
* Nixpkgs version: "17.03pre100133.a14217e"

**Comment by volth** [comment]
BTW, if ```jre8``` works well on macos, why it is the only nix target with default ```jre7``` ?

**Comment by copumpkin** [comment]
Almost definitely no reason to default to 7 on macOS (perhaps blame will show why it's like that?). A PR would be welcome!

**Comment by pSub** [comment]
This was already fixed in #22459.
